mParticle-Apple-SDK/Consent/MPCCPAConsent.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  MPCCPAConsent.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 9/17/24.
+//
+
+import Foundation
+
+/**
+ * Record of consent under the CCPA.
+ */
+@objc public final class MPCCPAConsent: NSObject, NSCopying {
+    
+    /**
+    * Whether the user consented to data collection
+    This should be set to false if the user has opted out of data sharing under the CCPA.
+    */
+    @objc public var consented: Bool = false
+    
+    /**
+    * The data collection document to which the user consented or did not consent
+    */
+    @objc public var document: String?
+    
+    /**
+    * Timestamp when the user was prompted for consent
+    */
+    @objc public var timestamp: Date
```

Comment:
```suggestion
    @objc public var timestamp = Date()
```

I'll add this to the style guide, but we should initialize immutable properties inline rather than in the init method.

Reply:
Also this is a great example of the preference to remove the explicit type from the declaration when initializing inline as it's usually very repetitive, i.e. `var timestamp: Date = Date()` vs `var timestamp = Date()`.